react/src/components/GeneSearch.tsx

```diff
@@ -57,16 +62,35 @@ const GeneSearch: React.FC<GeneSearchProps> = ({ assembly, geneName, onChange, o
     const formatAutocompleteOptions = useCallback(
         (autocompleteResults: AutocompleteResults) =>
             (autocompleteResults.autocompleteResults.hits || [])
-                .filter(hit => !!hit.ensembl?.gene && !!hit.genomic_pos && !!hit.genomic_pos_hg19)
-                .map((hit, i) => ({
-                    value: {
-                        name: hit.symbol.toUpperCase(),
-                        ensemblId: hit.ensembl?.gene,
-                        position: getPosition(assembly, hit),
-                    },
-                    id: i,
-                    label: hit.symbol.toUpperCase(),
-                })),
+                .filter(hit => !!hit.ensembl && !!hit.genomic_pos && !!hit.genomic_pos_hg19)
+                .map(hit => {
+                    const { symbol, ...rest } = hit;
+
+                    const ensembl = [rest.ensembl].flat();
+                    const genomic_pos = [rest.genomic_pos].flat();
+                    const genomic_pos_hg19 = [rest.genomic_pos_hg19].flat();
+                    return {
+                        symbol,
+                        ensembl,
+                        genomic_pos,
+                        genomic_pos_hg19,
+                    };
+                })
+                .map((hit, i) =>
+                    hit.ensembl.map((e, eid) => ({
+                        value: {
+                            name: hit.symbol.toUpperCase(),
+                            ensemblId: e.gene,
+                            position: getPosition(assembly, {
+                                genomic_pos: hit.genomic_pos[eid],
+                                genomic_pos_hg19: hit.genomic_pos_hg19[eid],
+                            }),
+                        },
+                        id: i + eid,
+                        label: isMultipleGenes(hit.ensembl) ? e.gene : hit.symbol.toUpperCase(),
+                    }))
```

Comment:
You could do this in the prior map call to only loop through the hits once, unless you plan to reuse the separate functions.

Reply:
Sounds good!